tools/flakes/detect_flakes.py

```diff
@@ -13,92 +13,251 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Detect new flakes introduced in the last 24h hours with respect to the
-previous six days"""
+"""Detect new flakes and create issues for them"""
 
 from __future__ import absolute_import
 from __future__ import division
 from __future__ import print_function
 
 import datetime
+import json
+import logging
 import os
+import pprint
 import sys
-import logging
-logging.basicConfig(format='%(asctime)s %(message)s')
+import urllib2
+from collections import namedtuple
 
 gcp_utils_dir = os.path.abspath(
     os.path.join(os.path.dirname(__file__), '../gcp/utils'))
 sys.path.append(gcp_utils_dir)
 
 import big_query_utils
 
-def print_table(table):
-    kokoro_base_url = 'https://kokoro.corp.google.com/job/'
-    for k, v in table.items():
-      job_name = v[0]
-      build_id = v[1]
-      ts = int(float(v[2]))
-      # TODO(dgq): timezone handling is wrong. We need to determine the timezone
-      # of the computer running this script.
-      human_ts = datetime.datetime.utcfromtimestamp(ts).strftime('%Y-%m-%d %H:%M:%S PDT')
-      job_path = '{}/{}'.format('/job/'.join(job_name.split('/')), build_id)
-      full_kokoro_url = kokoro_base_url + job_path
-      print("Test: {}, Timestamp: {}, url: {}\n".format(k, human_ts, full_kokoro_url))
+GH_ISSUES_URL = 'https://api.github.com/repos/grpc/grpc/issues'
+KOKORO_BASE_URL = 'https://kokoro2.corp.google.com/job/'
+
+def gh(url, data=None):
```

Comment:
Perhaps it would be fine for the tool to only print the verbatim info that should be added to a newly created issue. That would make filing issues much easier, but there would still be an option to only file some of them  (deduping, initial triage, etc.). You probably want to visit the newly created issues anyway and assign some labels to them and choose an assignee and perhaps add some initial observations and ideas. And it would also prevent issue inflation to some extent.

Reply:
I've added an interactive prompt asking, for each newly detected failure, whether to create a new issue. They are all created under the "infra/New Failure" label and one can optionally assign them as well.